test/lib/ansible_test/_data/sanity/code-smell/runtime-metadata.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+"""Schema validation of ansible-base's ansible_builtin_runtime.yml and collection's meta/runtime.yml"""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import datetime
+import os
+import re
+import sys
+import yaml
+
+from voluptuous import Any, MultipleInvalid, PREVENT_EXTRA
+from voluptuous import Required, Schema, Invalid
+from voluptuous.humanize import humanize_error
+
+from ansible.module_utils.six import string_types
+
+
+def isodate(v):
+    """Validate a datetime.date or ISO 8601 date string."""
+    # datetime.date objects come from YAML dates, these are ok
+    if isinstance(v, datetime.date):
+        return v
+    # make sure we have a string
+    msg = 'Expected ISO 8601 date string (YYYY-MM-DD), or YAML date'
+    if not isinstance(v, string_types):
+        raise Invalid(msg)
+    try:
+        datetime.datetime.strptime(v, '%Y-%m-%d').date()
+    except ValueError:
+        raise Invalid(msg)
+    return v
+
+
+def main():
+    """Validate runtime"""
+    ansible_builtin_runtime = 'lib/ansible/config/ansible_builtin_runtime.yml'
+    collection_legacy_file = 'meta/routing.yml'
+    collection_runtime_file = 'meta/runtime.yml'
+    path = ''
+
+    # Check if the correct routing files can be found
+
+    if os.path.isfile(ansible_builtin_runtime):
+        # We are in side an ansible-base checkout, so validate ansible_builtin_runtime
+        path = ansible_builtin_runtime
+    elif os.path.isfile(collection_runtime_file):
+        # We are in a collection, so validate meta/routing.yml
+        path = collection_runtime_file
+
+    if os.path.isfile(collection_legacy_file):
+        # Collection contains old name, renamed in #67684
+        print('%s:%d:%d: %s' % (collection_legacy_file, 0, 0, ("Should be called '%s'" % collection_runtime_file)))
+
+    if not path:
+        # In a collection, though no meta/routing
+        sys.exit(0)
+
+    try:
+        with open(path, 'r') as f_path:
+            routing = yaml.safe_load(f_path)
+    except yaml.error.MarkedYAMLError as ex:
+        print('%s:%d:%d: YAML load failed: %s' % (path, ex.context_mark.line +
+                                                  1, ex.context_mark.column + 1, re.sub(r'\s+', ' ', str(ex))))
+        sys.exit()
```

Comment:
Be explicit with exits:

```suggestion
        sys.exit(0)
```